service/lib/dinstaller/storage/proposal.rb

```diff
@@ -28,32 +27,59 @@ module DInstaller
   module Storage
     # Backend class to calculate a storage proposal
     class Proposal
-      include Singleton
-
       class NoProposalError < StandardError; end
 
+      # Constructor
+      #
+      # @param logger [Logger]
+      def initialize(logger)
+        @logger = logger
+      end
+
+      # Available devices for installation
+      #
+      # @return [Array<Y2Storage::Device>]
       def available_devices
         disk_analyzer.candidate_disks
       end
 
+      # Name of devices where to perform the installation
+      #
+      # @raise [NoProposalError] if no proposal yet
+      #
+      # @return [Array<String>]
       def candidate_devices
         raise NoProposalError unless proposal
 
         proposal.settings.candidate_devices
       end
 
+      # Whether the proposal should create LVM devices
+      #
+      # @raise [NoProposalError] if no proposal yet
+      #
+      # @return [Boolean]
       def lvm?
         raise NoProposalError unless proposal
 
         proposal.settings.use_lvm
       end
 
+      # Whether the proposal was successfully calculated
+      #
+      # @raise [NoProposalError] if no proposal yet
+      #
+      # @return [Boolean]
       def success?
         raise NoProposalError unless proposal
 
         !proposal.failed?
       end
 
+      # Calculates a new proposal
+      #
+      # @param settings [Hash] settings to calculate the proposal
+      #   (e.g., { "LVM" => true, "CandidateDevices" => ["/dev/sda"]})
```

Comment:
I wonder if we should use downcase keys and keep the conversation from D-Bus style in the corresponding D-Bus class. However, it is something we could do later (when we start writing RSpec tests).

Reply:
Hehe, that is exactly what it does! Sorry, documentation is wrong.